packages/protocol/contracts/common/UsingPrecompiles.sol

```diff
@@ -2,9 +2,11 @@ pragma solidity ^0.5.13;
 
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import "../common/interfaces/ICeloVersionedContract.sol";
+import "../common/UsingRegistry.sol";
+import "../common/interfaces/IEpochManager.sol";
 import "../../contracts-0.8/common/IsL2Check.sol";
 
-contract UsingPrecompiles is IsL2Check {
+contract UsingPrecompiles is IsL2Check, UsingRegistry {
```

Comment:
UsingRegistry has a registry variable that needs to be set, if we add the dependency here then if the childs don't have UseRegistry they would break.

Reply:
Updating it here would also trigger redeployments of every child

Reply:
Can UsingRegistryV2 be used instead? That uses only `constants`. Or is the Registry address different on L2?

Reply:
Yep, unfortunately it not possible to add new inherited smart contract with storage variable to already deployed smart contract (there is very high probability that this change will break storage)

Reply:
using `PrecompileOverride` contract, as discussed [here](https://github.com/celo-org/celo-blockchain-planning/issues/642#issuecomment-2397407994)